analyses/cell-type-ewings/scripts/utils/tumor-validation-helpers.R

```diff
@@ -0,0 +1,200 @@
+  marker_genes_df <- readr::read_tsv(marker_genes_file) |> 
+    # account for genes being from multiple sources
+    dplyr::select(cell_type, ensembl_gene_id, gene_symbol) |> 
+    dplyr::distinct()
+  
+  # get list of marker genes to use 
+  marker_genes <- marker_genes_df |> 
+    dplyr::filter(cell_type == "tumor") |> 
+    dplyr::pull(ensembl_gene_id)
+  
+  # get the gene expression counts for all marker genes
+  marker_gene_exp <- logcounts(sce[marker_genes, ]) |>
+    as.matrix() |>
+    t() |>
+    as.data.frame() |>
+    tibble::rownames_to_column("barcodes")
+  
+  # calculate sum of the scaled gene expression values for plotting 
+  plot_markers_df <- classification_df |> 
+    # add in marker gene expression to dataframe 
+    dplyr::left_join(marker_gene_exp, by = "barcodes") |> 
+    # combine all genes into a single column for easy faceting
+    tidyr::pivot_longer(
+      cols = starts_with("ENSG"),
+      names_to = "ensembl_gene_id",
+      values_to = "gene_expression"
+    ) |> 
+    # join with marker gene df to get gene symbols for plotting 
+    dplyr::left_join(marker_genes_df, by = c("ensembl_gene_id")) |> 
+    # keep only gene symbols, barcodes, and classification columns 
+    dplyr::select(barcodes, gene_symbol, gene_expression, ends_with("classification")) |> 
+    unique() |> 
+    dplyr::group_by(gene_symbol) |>
+    # get z-scores for each gene
+    dplyr::mutate(transformed_gene_expression = scale(gene_expression)[, 1]) |> 
+    dplyr::ungroup() |> 
+    dplyr::group_by(barcodes) |> 
+    # add all the scores together 
+    dplyr::mutate(sum_exp = sum(transformed_gene_expression)) |>
+    dplyr::ungroup()
+  
+  return(plot_markers_df)
+}
+
+# Heatmaps ---------------------------------------------------------------------
+
+# create a heatmap where rows are marker genes or gene sets and columns are cells 
+# this function adds a column annotation since cells are columns 
+plot_gene_heatmap <- function(
+    df, 
+    row_title = "",
+    legend_title = "", 
+    annotation = NULL
+){
+  
+  # plot heatmap of marker genes 
+  heatmap <- ComplexHeatmap::Heatmap(
+    df,
+    # set the color scale based on min and max values
+    col = circlize::colorRamp2(seq(min(df), max(df), length = 2), colors = c("white", "#00274C")),
+    border = TRUE,
+    ## Row parameters
+    cluster_rows = TRUE,
+    row_title = row_title,
+    row_title_gp = grid::gpar(fontsize = 12),
+    row_title_side = "left",
+    row_names_side = "left",
+    row_dend_side = "right",
+    row_names_gp = grid::gpar(fontsize = 10),
+    ## Column parameters
+    cluster_columns = TRUE,
+    show_column_names = FALSE,
+    bottom_annotation = annotation,
+    heatmap_legend_param = list(
+      title = legend_title
+    )
+  )
+  
+  return(heatmap)
+}
+
+# create a heatmap where rows are cells and columns are chromosomes 
+# this function adds a row annotation since cells are rows 
+plot_cnv_heatmap <- function(
+    df,
+    cnv_col, 
```

Comment:
Maybe something a bit more descriptive? Probably worth a comment too.
```suggestion
    cnv_col_prefix, 
```